service/worker/pernamespaceworker.go

```diff
@@ -369,10 +410,39 @@ func (w *perNamespaceWorker) startWorker(
 	return client, sdkworker, nil
 }
 
+func (w *perNamespaceWorker) onFatalError(err error) {
+	// clean up worker and client
+	w.stopWorker()
+
+	switch err.(type) {
+	case *serviceerror.NamespaceNotFound:
+		// if this is a NamespaceNotFound, we should retry
+		w.handleError(err)
```

Comment:
Will this keep retry forever? Why not just add a simple retry around StartWorker with expiration time like 20s?

Reply:
Oh, I see the problem. The error is not from Start, but from polling and only able to hook up with the OnFatalError hook. It make sense.

Reply:
> Will this keep retry forever?

It's using the default `backoff.NewExponentialRetryPolicy` with a 1s start. The defaults are 10s max interval, 1 minute expiration. So it will eventually stop. For this specific case (new namespace created), it should succeed within 10s since that's the default refresh interval and all nodes will see it by then.

After it hits max retries, it won't try again until it sees that a namespace is not assigned to this node anymore, and then assigned back to it.